# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/picture_graph.rs
# Commit: 9fa6f71111d0
# Full Hash: 9fa6f71111d0d77477abc4502b721d8d4ba23604
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2022-02-20 21:58:21
# Regressor Bug: 1749380
# File Overlap Count: 1
# Description:
#   Bug 1749380 - Part 1 - Improve how WR handles bounding rects for off-screen surfaces r=gfx-reviewers
#   
#   This patch introduces a number of subtle but important changes
#   to how we deal with off-screen surfaces. The overall goals are:
#   
# ==============================================================================

diff -r 336a2aa0a29e -r 9fa6f71111d0 gfx/wr/webrender/src/picture_graph.rs
--- a/gfx/wr/webrender/src/picture_graph.rs	Sat Feb 19 23:27:02 2022 +0000
+++ b/gfx/wr/webrender/src/picture_graph.rs	Sun Feb 20 18:56:54 2022 +0000
@@ -106,12 +106,18 @@
 
                 let info = &mut self.pic_info[pic_index.0];
 
-                info.surface_index = Some(pictures[pic_index.0].assign_surface(
+                match pictures[pic_index.0].assign_surface(
                     frame_context,
                     tile_caches,
-                    parent_surface_index,
                     surfaces,
-                ));
+                ) {
+                    Some(surface_index) => {
+                        info.surface_index = Some(surface_index);
+                    }
+                    None => {
+                        info.surface_index = Some(parent_surface_index.unwrap());
+                    }
+                }
             }
         }
     }